### ISSUE TITLE
possibility of UWP version?

### ISSUE THREAD

**Issue by scott-adamson1975** [ytmdesktop/ytmdesktop#1118]
### Consent

- [X] I verified that there is no open/closed issue for the same subject.
- [X] I understand that **YTMDesktop have NO affiliation with Google or YouTube**

### Feature

possibility of a UWP packaged version for use on xbox?

**Comment by mingjun97** [comment]
https://www.electronjs.org/docs/latest/tutorial/windows-store-guide

Looks feasible

**Comment by yoshiask** [comment]
Those docs don't describe what the CLI is actually doing. It's not performing a special recompile of your app, all it does is package the typical Win32 executable, with all its dependencies, in the APPX package format. It's still a Win32 app, just packaged, which is all that the MS Store requires. While this is still a good idea, it does not address the title of this issue.

**Comment by Alipoodle** [comment]
A few things to discuss here.

1) This project is called YTM Desktop and intended for Desktop Enviroment.
  We're not really a thing for all Windows platforms.
  
2) Mingjun's comment on "Looks feasible" was for a quick look into the posibility of it, VIA searching up `UWP Electron` or the likes. No reasearch has gone into this.
   This was never a full in depth look into it.

3) Other things which strike me as not very fesable from the original issue:
- the user is asking to get this made to run on an Xbox rather than run for Windows platform. There's a number of things which I suspect will have issues running in a far more locked down and containerised platform. (Technically some bits will work, but I don't see much expanstion the app will do which will be provided inside of UWP enviroment.
- To my minimal knowledge of Xbox Development (which is mainly from some emulation videos) this would only truly work as an app which would work as only a Foreground application to have to be displayed on the screen in the Development mode, and not in the standard Xbox Enviroment. (This requires you to pay a $25 developer fee)
More info here: https://learn.microsoft.com/en-us/windows/uwp/xbox-apps/devkit-activation


I feel like this is something which can be considered but a very unlikely task at hand. 😅
Given that the only known way of packaging up project is to use electron-windows-store which is just containerising the .exe and not making a REAL .appx then this might be a project which won't work for the user's original intension.

**Comment by Hershey025** [comment]
can this be uploaded to the microsoft store as a uwp app for windows 10 and 11?

**Comment by scott-adamson1975** [comment]
1) well, i've already broken that rule.... im using it on a steamdeck in
steam os... 😜

2) if spotify can do it, and deezer i believe it should be "feasible"

3) a) yes of course some bits will need abstracting for the various
platforms, you are supporting steamdecks linux multimedia controls, is that
not already an abstraction?
    b) not true, i believe the same linux/windows media controls hook rules
apply, just look at VLC, it can be used as a background player, and its UWP
app, and if an app is submitted through the proper channels once accepted
any retail box can download it from the store, i believe that submissions
to the microsoft store can be tagged as tablet, desktop or console... ala
VLC

**Comment by Alipoodle** [comment]
1) Steam deck is still running on a Desktop OS, and to many people would be still deemed a Desktop, just a portable one more intended for gaming.

2) As per consents:
> I understand that YTMDesktop have NO affiliation with Google or YouTube

We are not YouTube ourselves... We're not producing the original software, we're wrapping around  the existing YTM which already exists in the web, and built our own platform for it.

3)
A) We support Linux because in the space of Music and Development it was very little work to add a Linux build into it... 
There's a few features we have added for Linux specifically as well but, that's more of a platform where people are likely to use YTM.

B) We would need to get the YTM App onto the store, which is EXTREMELY unlikely given the fact that we are not the soul owners of the YTM license or anything, we are not afililated with them.
And there's difference between VLC, Spotify and Deezer VS A Community project.


**I'm not here to just down right destory this idea for a UWP version, or Xbox one.**
But as mentioend this is not a REAL build of a UWP and just a wrapper of the .exe in a UWP format. This would not function on an Xbox, and would be a Windows Desktop specific format.
We could possibly get it onto the Microsoft store but no guarantee that'd even consider us for a number of reasons nor would it be an optimal build to make for how many people would actually download & use it.

**Comment by yoshiask** [comment]
I understand the reluctance for UWP. However, I don't own an Xbox or Hololens (and of course I don't expect W10M support), and even on Windows Desktop I'd prefer a UWP client over an Electron one.

You don't necessarily have to publish to the Store. The only stipulation is that UWP apps have to be packaged and subsequently signed, which usually means Store publishing but there are other ways. In fact, if you're not afraid to ask users to install your self-cert, you don't need to pay for signing at all and just use VS's output.

The Store also accepts third-party clients. I wouldn't expect that to be an issue in the slightest, especially since all you're doing is embedding the official player without sidestepping any DRM or premium features.

**Comment by Hershey025** [comment]
yes also gpm desktop player was available on the store and you are not giving any premium features for free you don't block ads.

**Comment by Alipoodle** [comment]
TO BE CLEAR:
### I'm not saying no to a UWP for Windows / Putting into the store.

However Looking more into this while working on something else, we would need code signing which is something that this project currently doesn't have as it's an expensive thing for a community ran project which we do for free. 😅

An Xbox Build of this application (Which is the original poster's request) is not possible.
